plugins/scaffolder/src/api.ts

```diff
@@ -245,22 +242,22 @@ export class ScaffolderClient implements ScaffolderApi {
             }
           };
 
-          const eventSource = new EventSourcePolyfill(url, {
-            withCredentials: true,
-            headers: credentials?.token
-              ? { Authorization: `Bearer ${credentials.token}` }
-              : {},
-          });
-          eventSource.addEventListener('log', processEvent);
-          eventSource.addEventListener('recovered', processEvent);
-          eventSource.addEventListener('cancelled', processEvent);
-          eventSource.addEventListener('completion', (event: any) => {
-            processEvent(event);
-            eventSource.close();
-            subscriber.complete();
-          });
-          eventSource.addEventListener('error', event => {
-            subscriber.error(event);
+          fetchEventSource(url, {
```

Comment:
Does this automatically cleanup? We seem to missing the `eventSource.close` from before.

Reply:
I assume it does. It has an `onclose` hook, but no explicit call to close in the documentation: https://github.com/Azure/fetch-event-source?tab=readme-ov-file#usage

Reply:
updated

Reply:
I wonder if we should have the option of passing in an `AbortController`, and default to creating one if one is not provided? Maybe can save that for a followup though..